src/app.rs

```diff
@@ -270,7 +282,12 @@ impl App {
                 self.csv_table_state.reset_buffer();
             }
             Control::FilterColumns(s) => {
-                let re = Regex::new(s.as_str());
+                let lower_s = s.to_lowercase();
+                let re = if self.ignore_case && lower_s.starts_with(s) {
+                    Regex::new(&format!("(?i){}", s.as_str()))
+                } else {
+                    Regex::new(s.as_str())
+                };
```

Comment:
I think it would be nice if this can be refactored as a helper function and reused here and above.

Reply:
Good idea! Will do so...

Reply:
Hi @YS-L , I implemented the helper as per your feedback.